src/python/pants/backend/helm/target_types.py

```diff
@@ -482,12 +484,56 @@ class HelmDeploymentValuesField(DictStringToStringField):
         """
     )
 
+    @memoized_method
+    @deprecated("2.19.9.dev0", start_version="2.18.0rc0")
```

Comment:
I think we usually target the removal for a `.0.dev0` release, also I think we would prefer to introduce the deprecation also on the first dev version, rather than when we go into rc, so in this case perhaps:
```suggestion
    @deprecated("2.19.0.dev0", start_version="2.18.0.dev0")
```

Reply:
won't these deprecations fire regardless of if they are used or not?

Reply:
put this one mostly to cover the case in which someone has decided to use the method as an API in a custom plugin.

The deprecation in the method implementation (the one based on the interpolated values) would fire to warn about the deprecated syntax in the `BUILD` files.

There is indeed overlap between them although still a slight difference between the use cases.